cub/test/ptx-json/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+if ("${CMAKE_CXX_STANDARD}" STREQUAL "" OR "${CMAKE_CXX_STANDARD}" LESS "20")
+    return()
+endif()
+
+cccl_get_json()
+
+add_executable(ptx-json-test-filter
+  filter.cpp
+)
+cccl_configure_target(ptx-json-test-filter DIALECT 20)
+target_link_libraries(ptx-json-test-filter
+  nlohmann_json::nlohmann_json
+  CUDA::cudart
+  Thrust::Thrust
+  CUB::CUB
+)
+
+function(cub_detail_ptx_json_add_test target_name_var source)
+  string(REGEX REPLACE "test_([^.]*).cu" "cub.detail.ptx_json.test.\\1.ptx" target_name "${source}")
+  set(target_name_var ${target_name} PARENT_SCOPE)
+
+  add_library(${target_name} OBJECT
+    "${source}"
+  )
+  cccl_configure_target(${target_name} DIALECT 20)
+
+  target_link_libraries(${target_name} PRIVATE
+    CUB::CUB
+  )
+  set_target_properties(${target_name} PROPERTIES
+    CUDA_PTX_COMPILATION ON
+    CUDA_ARCHITECTURES 90
+  )
+
+  add_test(NAME ${target_name}
+    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/dump_and_check.bash
```

Comment:
question: should this only work on linux?

Reply:
As written yes. I can try to look into making it portable.

Reply:
I'm less concerned about portability than breaking windows builds. Maybe just return from this file if it's not Linux is good enough for now.